### PR TITLE
added useContext, refactor forms.

### PR DESCRIPTION
**Pull request by honcharuk96**

### CODE REVIEW

**Comment by honcharuk96** [comment]
@Edward-Fedoruk ,I will add support navigation with `React-router` after learning it in more detail (Task 8)

File: src/App.js [hunk 1]
```diff
@@ -1,20 +1,62 @@
...
+export const AppContext = createContext(null);
+const App = () => {
+  const [posters, setPosters] = useState([]);
+  const [countPosters, setCountPosters] = useState(0);
+  const [selectedPoster, setSelectedPoster] = useState(false);
+  const [submitForm, setSubmitForm] = useState({ form: null });
+  const [activeNav, setActiveNav] = useState(() => links[0]);
+
+  useEffect(async () => {
+    const { count, posters } = activeNav === links[0] ? await getPosters() : await getPostersByCategory(activeNav.text);
```
Comment (Edward-Fedoruk) [inline src/App.js:28]: better to use if else here. More readable

File: src/App.js [hunk 2]
```diff
@@ -1,20 +1,62 @@
...
+    setPosters(posters);
+    setCountPosters(count);
+  }, [activeNav, submitForm]);
+
+  const triggerPosterService = serviceName => setSubmitForm(serviceName);
```
Comment (Edward-Fedoruk) [inline src/App.js:33]: this is the same as 
const triggerPosterService = setSubmitForm;
Think about it) No need for wrapper and variable

File: src/App.js [hunk 3]
```diff
@@ -1,20 +1,62 @@
...
+
+  const setSelectedPosterId = posterId => setSelectedPoster(posterId);
```
Comment (Edward-Fedoruk) [inline src/App.js:35]: same here

File: src/global/components/redButton/redButton.component.js [hunk 4]
```diff
@@ -3,12 +3,12 @@ import PropTypes from 'prop-types';
...
-    </>
-  );
+  <>
+    <RedButton revertColor={revertColor} onClick={click}>
+      {text}
+    </RedButton>
+  </>
+);
 
 export default React.memo(RedButtonComponent);
```
Comment (Edward-Fedoruk) [inline src/global/components/redButton/redButton.component.js:13]: why memo here?? Please read the suggested article about performance

File: src/postersBlock/posterForm/addForm.component.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            required
+            pattern={'https://.*'}
+            type={'url'}
+            name={'poster_path'}
+            value={posterFormData.poster_path}
+            onChange={onChangeField}
+          />
+        </Label>
+        <Label key={'vote_average'}>
+          {'rating'}
```
Comment (Edward-Fedoruk) [inline src/postersBlock/posterForm/addForm.component.js:71]: why not just  `rating`

File: src/postersBlock/posterForm/addForm.component.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+            min={0}
+            type={'number'}
+            name={'runtime'}
+            value={posterFormData.runtime}
+            onChange={onChangeField}
+          />
+        </Label>
+        <Label key={'overview'} fullWidth>
+          {'overview'}
+          <TextArea name={'overview'} value={posterFormData.overview} onChange={onChangeField}>
```
Comment (Edward-Fedoruk) [inline src/postersBlock/posterForm/addForm.component.js:101]: name="overview"

File: src/header/header.component.js [hunk 7]
```diff
@@ -1,24 +1,46 @@
...
     <>
-      <Header>
-        <TopMenu>
-          <LogoComponent />
-          <ButtonMovie onClick={() => changeStateModal()} />
-          {showAdd  && <PosterForm showAdd={showAdd}  changeStateModal={() => changeStateModal()} />}
-        </TopMenu>
-        <SearchForm />
-      </Header>
+      {posterIdForHeader.selectedPoster === false ? (
```
Comment (Edward-Fedoruk) [inline src/header/header.component.js:19]: posterIdForHeader.selectedPoster === false
can be simplified !posterIdForHeader.selectedPoster

File: src/header/header.component.js [hunk 8]
```diff
@@ -1,24 +1,46 @@
...
+      ) : (
+        <>
+          <Main>
+            <TopMenu style={{ padding: '30px 0' }}>
+              <LogoComponent />
+              <img
+                src={Search_Button}
+                alt={'close'}
+                onClick={() => posterIdForHeader.setSelectedPosterId(false)}
+                style={{ cursor: 'pointer' }}
```
Comment (Edward-Fedoruk) [inline src/header/header.component.js:37]: move it to styles file

File: src/header/header.component.js [hunk 9]
```diff
@@ -1,24 +1,46 @@
...
+            <LogoComponent />
+            <ButtonMovie onClick={setShowForm} />
+            {showAdd && <PosterForm showAdd={showAdd} changeStateModal={setShowForm} />}
+          </TopMenu>
+          <SearchForm />
+        </Header>
+      ) : (
+        <>
+          <Main>
+            <TopMenu style={{ padding: '30px 0' }}>
```
Comment (Edward-Fedoruk) [inline src/header/header.component.js:31]: move to styles file

File: src/header/header.component.js [hunk 10]
```diff
@@ -1,24 +1,46 @@
...
+        <Header>
+          <TopMenu>
+            <LogoComponent />
+            <ButtonMovie onClick={setShowForm} />
+            {showAdd && <PosterForm showAdd={showAdd} changeStateModal={setShowForm} />}
+          </TopMenu>
+          <SearchForm />
+        </Header>
+      ) : (
+        <>
```
Comment (Edward-Fedoruk) [inline src/header/header.component.js:29]: no need for react fragment here. Just remove it

File: src/App.js [hunk 11]
```diff
@@ -1,20 +1,62 @@
...
+
+  return (
+    <>
+      <ErrorBoundary>
+        <GlobalStyles />
+        <AppContext.Provider
+          value={{
+            postersInfo: { posters, countPosters },
+            nav: { links, setActiveNav, activeNav },
+            posterIdForHeader: {
```
Comment (Edward-Fedoruk) [inline src/App.js:45]: should be a better name. Very confusing

File: src/App.js [hunk 12]
```diff
@@ -1,20 +1,62 @@
...
+  { id: 2, text: 'comedy' },
+  { id: 3, text: 'horror' },
+  { id: 4, text: 'crime' },
+];
+
+export const AppContext = createContext(null);
+const App = () => {
+  const [posters, setPosters] = useState([]);
+  const [countPosters, setCountPosters] = useState(0);
+  const [selectedPoster, setSelectedPoster] = useState(false);
```
Comment (Edward-Fedoruk) [inline src/App.js:23]: for boolean state usual naming convention - `isPosterSelected`

File: src/App.js [hunk 13]
```diff
@@ -1,20 +1,62 @@
...
+  const [submitForm, setSubmitForm] = useState({ form: null });
```
Comment (Edward-Fedoruk) [inline src/App.js:24]: The name is confusing. because it reads as a verb. Verbs usually are function. Should be a noun.

File: src/App.js [hunk 14]
```diff
@@ -1,20 +1,62 @@
...
+  const [activeNav, setActiveNav] = useState(() => links[0]);
+
+  useEffect(async () => {
```
Comment (Edward-Fedoruk) [inline src/App.js:27]: using async as useEffect callback is anti-pattern
https://fernandoabolafio.medium.com/how-to-go-async-with-useeffect-ff74dff0eec6

File: src/postersBlock/postersBlock.component.js [hunk 15]
```diff
@@ -1,39 +1,16 @@
...
-      <Main>
-        <NavigationCategory links={links} setActive={setActiveNav} activeNav={activeNav} />
-        <PosterCount count={countPosters} />
-        <ListPosters posters={posters} />
-      </Main>
-    </>
-  );
-};
-
+const PostersBlock = () => (
```
Comment (Edward-Fedoruk) [inline src/postersBlock/postersBlock.component.js:7]: PostersArea would be better name

File: src/App.js [hunk 16]
```diff
@@ -1,20 +1,62 @@
...
+
+  const setSelectedPosterId = posterId => setSelectedPoster(posterId);
+
+  return (
+    <>
+      <ErrorBoundary>
+        <GlobalStyles />
+        <AppContext.Provider
+          value={{
+            postersInfo: { posters, countPosters },
```
Comment (Edward-Fedoruk) [inline src/App.js:43]: countPosters is a verb. Change it to be a noun

**Review by Edward-Fedoruk — APPROVED** [review]

**Pull request merged**